server/src/main/java/org/kaazing/gateway/server/context/resolve/DefaultConnectOptionsContext.java

```diff
@@ -114,6 +121,19 @@ public DefaultConnectOptionsContext(ServiceConnectOptionsType connectOptions, Se
                 }
             }
 
+            ServiceConnectOptionsType.HttpKeepalive.Enum alive = connectOptions.getHttpKeepalive();
+            if (alive != null) {
+                tmpHttpKeepaliveEnabled = alive != ServiceConnectOptionsType.HttpKeepalive.DISABLED;
+            }
+
+            String timeoutValue = connectOptions.getHttpKeepaliveTimeout();
+            if (timeoutValue != null) {
+                long val = Utils.parseTimeInterval(timeoutValue, TimeUnit.SECONDS);
+                if (val > 0) {
+                    tmpHttpKeepaliveTimeout = val;
+                }
+            }
+
         }
 
         // Set default values via service-defaults
```

Comment:
David recently made a change to allow service-defaults to include connect-options
I do see why you are using the tmpXxx variables: to allow the members to be final.